datalad/support/gitrepo.py

```diff
@@ -1363,9 +1363,17 @@ def _git_custom_command(self, files, cmd_str,
         -------
         stdout, stderr
         """
-        cmd = shlex.split(cmd_str + " " + " ".join(files), posix=not on_windows) \
-            if isinstance(cmd_str, string_types) \
-            else cmd_str + files
+
+        if isinstance(cmd_str, string_types):
+            if files and not cmd_str.strip().endwith(" --"):
+                cmd_str += " --"
+            cmd = shlex.split(cmd_str + " " + " ".join(files),
+                              posix=not on_windows)
+        else:
+            if files and cmd_str[-1] != '--':
```

Comment:
I would have
-Taken out appending files our of condition (it is a list, we know)
- composed command list with files
 - kept this check dedented, and checked not only the last one but if any has `--`

This would allow to catch when anyone specified `--` anywhere

Reply:
Okay. I agree.